### PR TITLE
[DowngradePhp72] Skip DowngradePregUnmatchedAsNullConstantRector on different flags in BitwiseOr

### LINKED ISSUE #6074: Incorrect behavior in DowngradePregUnmatchedAsNullConstantRector
# Bug Report

| Subject        | Details                                                         |
| :------------- | :---------------------------------------------------------------|
| Rector version | e.g. v0.10.4              |
| Installed as   | composer dependency                          |

Rule `DowngradePregUnmatchedAsNullConstantRector` is downgrading the code below, but it should not.

## Minimal PHP Code Causing Issue

I tried to create the issue via getrector.org/demo, but I couldn't, see: https://github.com/rectorphp/getrector.org/issues/304

This code is being downgraded, but it should not, since the flag in `preg_match_all` is `PREG_OFFSET_CAPTURE | PREG_SET_ORDER`, and the rule should be applied for `PREG_UNMATCHED_AS_NULL` only:

```diff
/**
 * Parses a route string that does not contain optional segments.
 */
private function parsePlaceholders($route) {
    if (!preg_match_all(
        '~' . self::VARIABLE_REGEX . '~x', $route, $matches,
        PREG_OFFSET_CAPTURE | PREG_SET_ORDER
    )) {
        return [$route];
    }
+    array_walk_recursive($matches, function (&$value) {
+        if ($value === '') {
+            $value = null;
+        }
+    });
    $offset = 0;
    $routeData = [];
    foreach ($matches as $set) {
        if ($set[0][1] > $offset) {
            $routeData[] = substr($route, $offset, $set[0][1] - $offset);
        }
        $routeData[] = [
            $set[1][0],
            isset($set[2]) ? trim($set[2][0]) : self::DEFAULT_DISPATCH_REGEX
        ];
        $offset = $set[0][1] + strlen($set[0][0]);
    }

    if ($offset != strlen($route)) {
        $routeData[] = substr($route, $offset);
    }

    return $routeData;
}
```

## Expected Behaviour

The rule should be skipped.

### PR DESCRIPTION
**Pull request by samsonasik**
Fixes #6074

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: rules/LeagueEvent/Rector/MethodCall/DispatchStringToObjectRector.php [hunk 1]
```diff
@@ -94,12 +97,7 @@ private function shouldSkip(MethodCall $methodCall): bool
         ])) {
             return false;
         }
-
-        if (! $this->getStaticType($methodCall->args[0]->value) instanceof StringType) {
-            return true;
-        }
-
-        return true;
+        return $this->getStaticType($methodCall->args[0]->value) instanceof StringType;
```
Comment (samsonasik) [inline rules/LeagueEvent/Rector/MethodCall/DispatchStringToObjectRector.php:100]: @lulco this seems always return true, should just return `! $this->nodeTypeResolver->isObjectTypes()` above cond.

**Comment by TomasVotruba** [comment]
Thank you :+1:

File: rules/LeagueEvent/Rector/MethodCall/DispatchStringToObjectRector.php (hunk 1 above)
Comment (samsonasik) [reply rules/LeagueEvent/Rector/MethodCall/DispatchStringToObjectRector.php:100]: ref the original code that always return true: https://github.com/rectorphp/rector/blob/4b71eaa53d16c9c8cf4a6a2eb1fe4bf6d8006c58/rules/LeagueEvent/Rector/MethodCall/DispatchStringToObjectRector.php#L96-L100

**Comment by leoloso** [comment]
Thank you 👍

**Pull request merged**